EstateReporting.IntegrationTests/Shared/SharedSteps.cs

```diff
@@ -0,0 +1,749 @@
+                // Cache the merchant id
+                estateDetails.AddMerchant(response.MerchantId, merchantName);
+
+                this.TestingContext.Logger.LogInformation($"Merchant {merchantName} created with Id {response.MerchantId} for Estate {estateDetails.EstateName}");
+            }
+
+            foreach (TableRow tableRow in table.Rows)
+            {
+                EstateDetails estateDetails = this.TestingContext.GetEstateDetails(tableRow);
+
+                String merchantName = SpecflowTableHelper.GetStringRowValue(tableRow, "MerchantName");
+
+                Guid merchantId = estateDetails.GetMerchantId(merchantName);
+
+                String token = this.TestingContext.AccessToken;
+                if (String.IsNullOrEmpty(estateDetails.AccessToken) == false)
+                {
+                    token = estateDetails.AccessToken;
+                }
+
+                await Retry.For(async () =>
+                                {
+                                    MerchantResponse merchant = await this.TestingContext.DockerHelper.EstateClient
+                                                                          .GetMerchant(token, estateDetails.EstateId, merchantId, CancellationToken.None)
+                                                                          .ConfigureAwait(false);
+
+                                    merchant.MerchantName.ShouldBe(merchantName);
+                                });
+            }
+        }
+        
+        [When(@"I perform the following transactions")]
+        public async Task WhenIPerformTheFollowingTransactions(Table table)
+        {
+            foreach (TableRow tableRow in table.Rows)
+            {
+                String merchantName = SpecflowTableHelper.GetStringRowValue(tableRow, "MerchantName");
+                String dateString = SpecflowTableHelper.GetStringRowValue(tableRow, "DateTime");
+                DateTime transactionDateTime = SpecflowTableHelper.GetDateForDateString(dateString, DateTime.Now);
+                // hack for UTC :|
+                transactionDateTime = transactionDateTime.AddHours(1);
+                String transactionNumber = SpecflowTableHelper.GetStringRowValue(tableRow, "TransactionNumber");
+                String transactionType = SpecflowTableHelper.GetStringRowValue(tableRow, "TransactionType");
+                String deviceIdentifier = SpecflowTableHelper.GetStringRowValue(tableRow, "DeviceIdentifier");
+                
+                EstateDetails estateDetails = this.TestingContext.GetEstateDetails(tableRow);
+                
+                // Lookup the merchant id
+                Guid merchantId = estateDetails.GetMerchantId(merchantName);
+                SerialisedMessage transactionResponse = null;
+                switch (transactionType)
+                {
+                    case "Logon":
+                        transactionResponse = await this.PerformLogonTransaction(estateDetails.EstateId,
```

Comment:
Codacy found an issue: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/stuart_ferguson1/EstateReporting/pullRequest?prid=6339123)